src/System.Security.Cryptography.X509Certificates/tests/ChainTests.cs

```diff
@@ -226,6 +226,88 @@ public static void BuildChainExtraStoreUntrustedRoot()
             }
         }
 
+        [Fact]
+        public static void BuildChainCustomTrustStore()
+        {
+            using (var testCert = new X509Certificate2(Path.Combine("TestData", "test.pfx"), TestData.ChainPfxPassword))
+            using (var microsoftDotCom = new X509Certificate2(TestData.MicrosoftDotComSslCertBytes))
+            using (var chainHolder = new ChainHolder())
+            {
+                // Normal case
+                X509Chain chain = chainHolder.Chain;
+                chain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
+                chain.ChainPolicy.VerificationTime = new DateTime(2015, 9, 22, 12, 25, 0);
+
+                Assert.True(chain.Build(microsoftDotCom));
+                Assert.Equal(3, chain.ChainElements.Count);
+                X509Certificate2 rootCert = chain.ChainElements[2].Certificate;
+                X509Certificate2 intermediateCert = chain.ChainElements[1].Certificate;
+
+                //Trust mode set to custom, but no certificates added
+                chain.ChainPolicy.TrustMode = X509ChainTrustMode.CustomRootTrust;
```

Comment:
Ideally each sub-case here would be on a different chain object, and possibly from a different method (or the same method driven by theory inputs).

Currently this method is not disposing the certificates created from each prior chain, and if something fails it won't be obvious in the logs what went wrong (too many Assert.Falses, too many Assert.Equals(3), too many Assert.Equal(UntrustedRoot))